Monika After Story/game/chess.rpy

```diff
@@ -2306,553 +1307,1562 @@ label mas_chess_go_ham_and_delete_everything:
+                MASChessDisplayableBase.BUTTON_WIDTH,
+                MASChessDisplayableBase.BUTTON_HEIGHT,
+                hover_sound=gui.hover_sound,
+                activate_sound=gui.activate_sound
+            )
 
-# quick version of game in progress
-label mas_chess_dlg_game_in_progress_end_quick:
-    m 1eua "Okay, [player], let's continue this game soon."
-    return
+            if practice_mode:
+                self._button_undo = MASButtonDisplayable.create_stb(
+                    _("Undo"),
+                    True,
+                    MASChessDisplayableBase.BUTTON_INDICATOR_X,
+                    MASChessDisplayableBase.DRAWN_BUTTON_Y_BOT,
+                    MASChessDisplayableBase.BUTTON_WIDTH,
+                    MASChessDisplayableBase.BUTTON_HEIGHT,
+                    hover_sound=gui.hover_sound,
+                    activate_sound=gui.activate_sound
+                )
 
-#### end dialogue blocks ######################################################
+                #Setup the visible buttons list
+                self._visible_buttons = [
+                    self._button_save,
+                    self._button_undo,
+                    self._button_giveup
+                ]
 
-# confirmation screen for chess
-screen mas_chess_confirm():
+            else:
+                self.visible_buttons = [
+                    self._button_save,
+                    self._button_giveup
+                ]
 
-    ## Ensure other screens do not get input while this screen is displayed.
-    modal True
+            self._visible_buttons_winner = [
+                self._button_done
+            ]
+
+            #Init the base displayable
+            super(MASChessDisplayable, self).__init__(
+                player_color,
+                pgn_game,
+                starting_fen,
+                buttons={
+                    "_button_save": {
+                        "conditional": (
+                            "not is_game_over "
+                            "and player_color == current_turn "
+                            "and board.fullmove_number > 1"
+                        )
+                    },
+                    "_button_giveup": {
+                        "conditional": (
+                            "not is_game_over "
+                            "and player_color == current_turn "
+                            "and board.fullmove_number > 1"
+                        )
+                    },
+                    "_button_done": {
+                        "conditional": "is_game_over is not None"
+                    },
+                    "_button_undo": {
+                        "conditional": (
+                            "not is_game_over "
+                            "and player_color == current_turn "
+                            "and board.fullmove_number > 0 "
+                            "and move_history"
+                        )
+                    }
+                },
```

Comment:
seeing this, I dont think its really worth it to generalize button conditions, especially since this is only for enabling/disabling buttons, and you still need make the button objects on the extended class to handle general flow. 
You could always write an abstract/not implemented `set_button_states` on the base class, and have the extended class implement the actual conditional checking. The only time the base class references `set_button_states` is in its constructor, which is barely enough to warrant passing and storing a generalized map for it. The extended classes can just call `set_button_states` in their constructors as needed.